src/pkg/exporter/project_collector.go

```diff
@@ -0,0 +1,239 @@
+	projectArtifactsSQL = `SELECT artifact.project_id, artifact.type AS artifact_type, COUNT(artifact.type) AS artifact_total
+	FROM project INNER JOIN artifact ON project.project_id=artifact.project_id
+	WHERE project.deleted=FALSE
+	GROUP BY artifact.project_id, type;`
+)
+var (
+	projectTotal = typedDesc{
+		desc:      newDescWithLables("", "project_total", "Total projects number", "public"),
+		valueType: prometheus.GaugeValue,
+	}
+	projectUsage = typedDesc{
+		desc:      newDescWithLables("", "project_usage_byte", "The used resource of a project", "project_id"),
+		valueType: prometheus.GaugeValue,
+	}
+	projectQuote = typedDesc{
+		desc:      newDescWithLables("", "project_quota_byte", "The quota of a project", "project_id"),
+		valueType: prometheus.GaugeValue,
+	}
+	projectRepoTotal = typedDesc{
+		desc:      newDescWithLables("", "project_repo_total", "Total project repos number", "project_id", "public"),
+		valueType: prometheus.GaugeValue,
+	}
+
+	projectMemberTotal = typedDesc{
+		desc:      newDescWithLables("", "project_member_total", "Total members number of a project", "project_id"),
+		valueType: prometheus.GaugeValue,
+	}
+	artifactPullTotal = typedDesc{
+		desc:      newDescWithLables("", "image_pulled", "The pull number of an image", "project_id"),
+		valueType: prometheus.GaugeValue,
+	}
+	projectArtifactTotal = typedDesc{
+		desc:      newDescWithLables("", "project_artifact_total", "Total project images number", "project_id", "public", "image_type"),
+		valueType: prometheus.GaugeValue,
+	}
+)
+
+var errProjectNotFound = errors.New("project not found")
+
+// NewProjectCollector ...
+func NewProjectCollector() *ProjectCollector {
+	return &ProjectCollector{}
+}
+
+// ProjectCollector ...
+type ProjectCollector struct{}
+
+// Describe implements prometheus.Collector
+func (hc *ProjectCollector) Describe(c chan<- *prometheus.Desc) {
+	c <- projectTotal.Desc()
+	c <- projectUsage.Desc()
+	c <- projectQuote.Desc()
+	c <- projectRepoTotal.Desc()
+	c <- projectMemberTotal.Desc()
+	c <- artifactPullTotal.Desc()
+	c <- projectArtifactTotal.Desc()
+}
+
+// Collect implements prometheus.Collector
+func (hc *ProjectCollector) Collect(c chan<- prometheus.Metric) {
+	overview := getProjectInfo()
+	for _, v := range overview.projectTotals {
+		c <- projectTotal.MustNewConstMetric(v.Count, getPublicValue(v.Public))
+	}
+	for _, p := range overview.ProjectMap {
+		c <- projectUsage.MustNewConstMetric(getQuotaValue(p.Usage), p.Name)
+		c <- projectQuote.MustNewConstMetric(getQuotaValue(p.Quota), p.Name)
+		c <- projectMemberTotal.MustNewConstMetric(p.MemberTotal, p.Name)
+		c <- projectRepoTotal.MustNewConstMetric(p.RepoTotal, p.Name, getPublicValue(p.Public))
+		c <- artifactPullTotal.MustNewConstMetric(p.PullTotal, p.Name)
+		c <- projectArtifactTotal.MustNewConstMetric(p.Artifact.ArtifactTotal, p.Name, getPublicValue(p.Public), p.Artifact.ArtifactType)
+	}
+}
+
+type projectOverviewInfo struct {
+	projectTotals []projectCount
```

Comment:
would it make more sense to render public, non-public explicitly as the projectCount array will always have 2 entries?